### DISCUSSION TITLE
How do i use so-called Offline mode？

### DISCUSSION THREAD

**Discussion by EasonChow001** [microsoft/onnxruntime#7176] [Other Q&A] [answered]
As illustrated in https://www.onnxruntime.ai/docs/resources/graph-optimizations.html#onlineoffline-mode, i expand my understanding about how to do a offline mode inference:
step1:  i should set `sessionOptions.setOptimizedModelFilePath()` and `sessionOptions.setOptimizationLevel()`
step2:  online mode,  run the session,so the optimized model will be save to OptimizedModelFilePath.
step3:  offline mode, remove the  `sessionOptions.setOptimizedModelFilePath()` and `sessionOptions.setOptimizationLevel()` in code, and createSession with the optimized model in OptimizedModelFilePath. And when we  run the session,  we will run the inference straightly skipping the optimization phase.

Am i right?

**Comment by pranavsharma** [comment, marked as answer]
Yes, this is correct.

**Reply by EasonChow001** [reply]
thx
